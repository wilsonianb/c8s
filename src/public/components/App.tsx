import * as React from 'react'
import ContainerForm from './ContainerForm'
import WebMonetizationLoader from './WebMonetizationLoader'

type AppProps = {
  codiusVersion: string
}

const App: React.FC<AppProps> = (props: AppProps) => {

  return (
    <WebMonetizationLoader>
      <ContainerForm codiusVersion={props.codiusVersion} />
    </WebMonetizationLoader>
  )
}

export default App
