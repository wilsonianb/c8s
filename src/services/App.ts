import { Injector } from 'reduct'
import HttpServer from './HttpServer'
import KubernetesClient from './KubernetesClient'
import SelfTest from './SelfTest'

import { create as createLogger } from '../common/log'
const log = createLogger('App')

export default class App {
  private httpServer: HttpServer
  private kubernetesClient: KubernetesClient
  private selfTest: SelfTest
  constructor (deps: Injector) {
    this.httpServer = deps(HttpServer)
    this.kubernetesClient = deps(KubernetesClient)
    this.selfTest = deps(SelfTest)
  }

  async start () {
    log.info('starting c8s...')
    await this.kubernetesClient.start()
    await this.httpServer.init()
    await this.httpServer.start()
    // this.selfTest.start()
  }
}
