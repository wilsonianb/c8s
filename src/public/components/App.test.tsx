/*!
 * @jest-environment jsdom
 */

import * as React from 'react'
import { mount, shallow } from 'enzyme'

import App from './App'

describe('<App />', () => {
  const codiusVersion: string = '0.0.1'

  test('matches snapshot', () => {
    const wrapper = mount(<App codiusVersion={codiusVersion} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('contains web monetization required message', () => {
    const wrapper = shallow(<App codiusVersion={codiusVersion} />)
    // expect(wrapper).toMatchElement('<WebMonetizationLoader><WithFormik(InnerForm) /></WebMonetizationLoader>')
  })

  // TODO: test with web monetization
})
