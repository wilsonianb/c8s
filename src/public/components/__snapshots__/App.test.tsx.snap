// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<App
  codiusVersion="0.0.1"
>
  <WebMonetizationLoader>
    <p>
      Sorry! You need 
      <a
        href="https://webmonetization.org"
      >
        Web Monetization
      </a>
       to view this content.
    </p>
  </WebMonetizationLoader>
</App>
`;
